theater-gorm/api/theater.go

```diff
@@ -0,0 +1,1345 @@
+	r.HandleFunc("/poster", api.createPoster).Methods("POST")
+	r.HandleFunc("/price", api.createPrice).Methods("POST")
+	r.HandleFunc("/role", api.createRole).Methods("POST")
+	r.HandleFunc("/schedule", api.createSchedule).Methods("POST")
+	r.HandleFunc("/sector", api.createSector).Methods("POST")
+	r.HandleFunc("/ticket", api.createTicket).Methods("POST")
+	r.HandleFunc("/user", api.createUser).Methods("POST")
+
+	r.HandleFunc("/account", api.updateAccount).Methods("PUT")
+	r.HandleFunc("/genre", api.updateGenre).Methods("PUT")
+	r.HandleFunc("/hall", api.updateHall).Methods("PUT")
+	r.HandleFunc("/location", api.updateLocation).Methods("PUT")
+	r.HandleFunc("/performance", api.updatePerformance).Methods("PUT")
+	r.HandleFunc("/place", api.updatePlace).Methods("PUT")
+	r.HandleFunc("/poster", api.updatePoster).Methods("PUT")
+	r.HandleFunc("/price", api.updatePrice).Methods("PUT")
+	r.HandleFunc("/role", api.updateRole).Methods("PUT")
+	r.HandleFunc("/schedule", api.updateSchedule).Methods("PUT")
+	r.HandleFunc("/sector", api.updateSector).Methods("PUT")
+	r.HandleFunc("/ticket", api.updateTicket).Methods("PUT")
+	r.HandleFunc("/user", api.updateUser).Methods("PUT")
+}
+
+func (a theaterAPI) getAllTickets(writer http.ResponseWriter, _ *http.Request) {
+	tickets, err := a.data.ReadAllTickets()
+	if err != nil {
+		_, err := writer.Write([]byte("got an error when tried to get tickets"))
+		if err != nil {
+			log.Println(err)
+		}
+	}
+	err = json.NewEncoder(writer).Encode(tickets)
+	if err != nil {
+		log.Println(err)
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+}
+
+func (a theaterAPI) getAllPosters(writer http.ResponseWriter, _ *http.Request) {
+	posters, err := a.data.ReadAllPosters()
+	if err != nil {
+		_, err := writer.Write([]byte("got an error when tried to get posters"))
+		if err != nil {
+			log.Println(err)
+		}
+	}
+	err = json.NewEncoder(writer).Encode(posters)
+	if err != nil {
+		log.Println(err)
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+}
+
+func (a theaterAPI) getAllUsers(writer http.ResponseWriter, request *http.Request) {
+	account := new(data.Account)
+
+	if n, err := strconv.Atoi(request.URL.Query().Get("idAccount")); err == nil {
+		account.Id = n
+	} else {
+		log.Printf("failed reading id: %s\n", err)
+		writer.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	users, err := a.data.ReadAllUsers(*account)
+	if err != nil {
+		_, err := writer.Write([]byte("got an error when tried to get users"))
+		if err != nil {
+			log.Println(err)
+		}
+	}
+	err = json.NewEncoder(writer).Encode(users)
+	if err != nil {
+		log.Println(err)
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+}
+
+func (a theaterAPI) getAccountById(writer http.ResponseWriter, request *http.Request) {
+	entity := new(data.Account)
+	if n, err := strconv.Atoi(request.URL.Query().Get("id")); err == nil {
+		entity.Id = n
+	} else {
+		log.Printf("failed reading id: %s\n", err)
```

Comment:
we should not add `\n` here. because log function does it automatically

Reply:
Done